train.py

```diff
@@ -44,32 +43,39 @@
     ObjectDetectionTask,
     RegressionTask,
     SemanticSegmentationTask,
+    SimCLRTask,
 )
 
-TASK_TO_MODULES_MAPPING: Dict[
-    str, Tuple[Type[LightningModule], Type[LightningDataModule]]
-] = {
-    "bigearthnet": (MultiLabelClassificationTask, BigEarthNetDataModule),
-    "byol": (BYOLTask, ChesapeakeCVPRDataModule),
-    "chesapeake_cvpr": (SemanticSegmentationTask, ChesapeakeCVPRDataModule),
-    "cowc_counting": (RegressionTask, COWCCountingDataModule),
-    "cyclone": (RegressionTask, TropicalCycloneDataModule),
-    "deepglobelandcover": (SemanticSegmentationTask, DeepGlobeLandCoverDataModule),
-    "eurosat": (ClassificationTask, EuroSATDataModule),
-    "etci2021": (SemanticSegmentationTask, ETCI2021DataModule),
-    "gid15": (SemanticSegmentationTask, GID15DataModule),
-    "inria": (SemanticSegmentationTask, InriaAerialImageLabelingDataModule),
-    "landcoverai": (SemanticSegmentationTask, LandCoverAIDataModule),
-    "loveda": (SemanticSegmentationTask, LoveDADataModule),
-    "naipchesapeake": (SemanticSegmentationTask, NAIPChesapeakeDataModule),
-    "nasa_marine_debris": (ObjectDetectionTask, NASAMarineDebrisDataModule),
-    "potsdam2d": (SemanticSegmentationTask, Potsdam2DDataModule),
-    "resisc45": (ClassificationTask, RESISC45DataModule),
-    "sen12ms": (SemanticSegmentationTask, SEN12MSDataModule),
-    "so2sat": (ClassificationTask, So2SatDataModule),
-    "spacenet1": (SemanticSegmentationTask, SpaceNet1DataModule),
-    "ucmerced": (ClassificationTask, UCMercedDataModule),
-    "vaihingen2d": (SemanticSegmentationTask, Vaihingen2DDataModule),
+DATASET_TO_DATAMODULES_MAPPING: Dict[str, Type[LightningDataModule]] = {
```

Comment:
Can we automatically infer these? They follow a consistent naming scheme.

Reply:
How would you infer these? I mentioned using hydra to instantiate from a config. Do you have any other suggestions?

Reply:
Can we use the full CamelCase name in the config? I'm fine with using hydra.

Reply:
It's possible to use either `torchgeo.trainers.ClassificationTask` or `ClassificationTask` in the config

Reply:
The next question is should we make a user only have to define `ClassificationTask` or prefer them to explicitly define the full path like `torchgeo.trainers.ClassificationTask`

Reply:
Ideally either one would work. The former because it's short, and the latter because a user may want to define their own trainer/datamodule. But if it's too much code, I would prefer the former for now.

Reply:
I'm going to use the latter because it basically will remove code but also not require us to always update the `train.py` script when we add a new datamodule, also allowing a user to define their own trainer/datamodule.

Reply:
Why would it require us to update `train.py`? Couldn't we just use `getattr(torchgeo.datamodules, "LandCoverAIDataModule")`?

Reply:
Yes that's right.